processor/deltatorateprocessor/factory_test.go

```diff
@@ -44,49 +46,42 @@ func TestCreateDefaultConfig(t *testing.T) {
 }
 
 func TestCreateProcessors(t *testing.T) {
-	tests := []struct {
-		configName   string
-		succeed      bool
-		errorMessage string
-	}{
-		{
-			configName: "config_full.yaml",
-			succeed:    true,
-		},
-	}
+	t.Parallel()
+
+	cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
+	require.NoError(t, err)
 
-	for _, test := range tests {
-		factories, err := componenttest.NopFactories()
-		assert.NoError(t, err)
+	for k := range cm.ToStringMap() {
+		// Check if all processor variations that are defined in test config can be actually created
+		t.Run(k, func(t *testing.T) {
+			factory := NewFactory()
+			cfg := factory.CreateDefaultConfig()
 
-		factory := NewFactory()
-		factories.Processors[typeStr] = factory
-		config, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", test.configName), factories)
-		assert.NoError(t, err)
+			var id string
+			parts := strings.Split(k, "/")
+			if len(parts) > 1 {
+				id = parts[1]
+			}
+			sub, err := cm.Sub(config.NewComponentIDWithName(typeStr, id).String())
```

Comment:
Same use `ComponentID.UnmarshalText`. But not sure you need to do that, just simply `sub, err := cm.Sub(k)`

Reply:
`sub, err := cm.Sub(k)` that is enough